lib/ansible/modules/network/nxos/nxos_user.py

```diff
@@ -221,7 +221,10 @@ def parse_password(data):
 
 
 def parse_roles(data):
-    configured_roles = data.get('TABLE_role')['ROW_role']
+    configured_roles = None
+    if 'TABLE_role' in data:
+        configured_roles = data.get('TABLE_role')['ROW_role']
```

Comment:
@mikewiebe `get` takes care of NoneType when the data doesn't have the key `TABLE_role`. It returns None. The error appears to be because the NoneType looking for the `ROW_role` key.
Checking for `TABLE_role` key when using `get` is not justified.
So Maybe remove `get` from this line? `configured_roles = data['TABLE_role']['ROW_role']`

Reply:
Sounds good.  I will remove the `get`.

Reply:
@trishnaguha Re-thinking this change.  I would prefer to leave the get as a safety measure.  It's not hurting anything and will prevent the module from generating an error.  I have seen cases where different Nexus versions use different key names.  Risk is small but still exists.

Reply:
ack. merging this in since this isn't a blocker.